airflow/operators/python.py

```diff
@@ -300,7 +304,7 @@ def __init__(
         self,
         *,
         python_callable: Callable,
-        requirements: Optional[Iterable[str]] = None,
+        requirements: Union[Iterable[str], str, None] = None,
```

Comment:
I think we usually tend to use `Optional` instead of `Union[..., None]` additionaly `Optional` marks the argument as... optional 😉 

```suggestion
        requirements: Optional[Union[Iterable[str], str]] = None,
```

Reply:
Makes sense, addressed!

Reply:
FWIW the two are equivalent, `Optional[X]` is just syntax sugar to `Union[X, None]` (and IIRC Mypy normalises the former to latter internally, not sure)